### PR TITLE
Taker Bot: order collector

### PR DESCRIPTION
**Pull request by JuaniRios**
## Motivation

Closes [RAI-8](https://linear.app/makeitrain/issue/RAI-8/phase-1-order-collector)

The direct taker bot needs to discover and track Raindex orders so it can determine which ones are worth taking. This PR introduces the first domain aggregate for the taker bot: `TrackedOrder`, which models the full lifecycle of a discovered order (Active -> Exhausted / Removed), along with the `OrderFilter` that decides whether an `AddOrderV3` event represents an order the bot should track.

### `TrackedOrder` CQRS aggregate (`crates/taker/src/tracked_order.rs`)

A full event-sourced aggregate using `st0x-event-sorcery` that tracks Raindex orders through their lifecycle:

- **States**: `Active` (discovered, available for taking), `Exhausted` (remaining output reached zero), `Removed` (owner removed the order)
- **Commands**: `Discover` (from `AddOrderV3` events), `RecordFill` (from `TakeOrderV3` events, ours or another taker's), `MarkRemoved` (from `RemoveOrderV3` events)
- **Idempotency**: Duplicate `Discover` on an existing aggregate is a no-op. Terminal states (`Exhausted`, `Removed`) reject all further commands
- **Fill tracking**: Partial fills decrement `remaining_output`; when it hits zero, both `OrderFilled` and `OrderExhausted` events are emitted. The `is_ours` flag distinguishes our fills from third-party takers
- **Domain types**: `OrderHash` (strongly-typed aggregate ID wrapping `B256`), `Scenario` (A = owner sells wtToken for USDC, B = owner buys wtToken with USDC), `OrderType` (placeholder for future classification step)

### `OrderFilter` (`crates/taker/src/tracked_order/filter.rs`)

Stateless filter that determines whether an `AddOrderV3` event should be tracked:

- **Excluded owner check**: Orders from the liquidity bot's own address are ignored (we don't take our own orders)
- **Token pair matching**: Checks all `(validInput, validOutput)` combinations from the `OrderV4` struct against configured USDC and wrapped token addresses. Only orders involving exactly one USDC side and one configured wtToken are supported
- **Scenario classification**: Determines whether the order is Scenario A (owner sells wtToken) or Scenario B (owner sells USDC) based on which side has USDC

### Test coverage

- **17 aggregate tests**: Discovery, partial/full fills, exhaustion, removal, terminal state rejection, event replay, idempotency, edge cases (fill exceeding remaining, fill/remove on nonexistent)
- **11 filter tests**: Excluded owner, Scenario A/B classification, unsupported pairs (unknown tokens, USDC-USDC, wtToken-wtToken), multiple IO combinations, empty arrays

## Checks

By submitting this for review, I'm confirming I've done the following:

- [x] added comprehensive test coverage for any changes in logic
- [x] made this PR as small as possible
- [x] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a change to the dashboard)

## Summary by CodeRabbit

* **New Features**
  * Full order-tracking runtime with historical backfill and live event processing for discovered orders.
* **Infrastructure**
  * Persistent event store and view table for reliable CQRS/event-sourced state.
  * Test-support tooling for local contract deployment and integration testing.
* **Tests**
  * Extensive integration and unit tests covering discovery, fills, removals, and backfill idempotency.

### CODE REVIEW

File: services/taker/src/order_collector/event_processor.rs [hunk 1]
```diff
@@ -0,0 +1,560 @@
...
+            .send(
+                &order_hash,
+                TrackedOrderCommand::Discover {
+                    owner,
+                    symbol,
+                    scenario,
+                    output_token,
+                    input_token,
+                    max_output: max_output_from_order(&event.order, scenario, output_token),
+                    block,
```
[the remark this answers is not in the document]
Comment (JuaniRios) [reply services/taker/src/order_collector/event_processor.rs]: Deferring to the Order Classification step (step 3 in the work plan). There's a detailed TODO at event_processor.rs:167-173 explaining why. While max_output is the U256::MAX placeholder, the raw-bits conversion is safe — exhaustion tracking is effectively disabled until classification provides real output bounds.

File: services/taker/src/lib.rs [hunk 2]
```diff
@@ -29,19 +49,291 @@ pub fn setup_tracing(log_level: &LogLevel) {
...
+                    }
+                }
+                Err(error) => error!("Error decoding RemoveOrderV3: {error}"),
+            },
+
+            Some(result) = take_stream.next() => match result {
+                Ok((event, log)) => {
+                    if processor.process_take_order(&event).await.is_ok() {
+                        advance_cursor(&cursor, &log).await;
+                    } else {
```
[the remark this answers is not in the document]
Comment (JuaniRios) [reply services/taker/src/lib.rs]: Same concern as the subscription/backfill overlap above — `Discover` is already idempotent, and `RecordFill` idempotency will land together with the subscription ordering fix in the next PR. The in-code TODO documents the plan.
